feat(gomod): support -e option (#20199)
renovatebot/renovate: docs/usage/golang.md

1. If Renovate finds an update, Renovate will update `go.mod` to the new value
1. Renovate runs `go get` to update the `go.sum` files
1. If the user has enabled the option `gomodUpdateImportPaths` in the [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions) array, then Renovate uses [mod](https://github.com/marwan-at-work/mod) to update import paths on major updates, which can update any Go source file

> 1. If the user has any of the available `gomodTidy` options ( e.g. `gomodTidy1.17` ) in the [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions), then Renovate runs `go mod tidy` with the respective options ( multiple options are allowed ).

Reviewer: ```suggestion
1. If the user has any of the available `gomodTidy` options (e.g. `gomodTidy1.17`) in the [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions), then Renovate runs `go mod tidy` with the respective options (multiple options are allowed).
```

1. If Renovate finds an update, Renovate will update `go.mod` to the new value
1. Renovate runs `go get` to update the `go.sum` files
1. If the user has enabled the option `gomodUpdateImportPaths` in the [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions) array, then Renovate uses [mod](https://github.com/marwan-at-work/mod) to update import paths on major updates, which can update any Go source file

> 1. If the user has enabled the option `gomodTidy`, `gomodTidy1.17` `gomodTidyE` in the [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions) array, then Renovate runs `go mod tidy`, `go mod tidy -compat=1.17` or `go mod tidy -e` respectively, which itself can update `go.mod` and `go.sum`.

Reviewer 2: ```suggestion
1. If the user has enabled the option `gomodTidy`, `gomodTidy1.17` or `gomodTidyE` in the [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions) array, then Renovate runs `go mod tidy`, `go mod tidy -compat=1.17` or `go mod tidy -e` respectively, which itself can update `go.mod` and `go.sum`.
```

Reviewer: ```suggestion
1. If the user has enabled the option `gomodTidy`, `gomodTidy1.17` or `gomodTidyE` in the [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions) array, then Renovate runs `go mod tidy`, `go mod tidy -compat=1.17` or `go mod tidy -e` respectively, which itself can update `go.mod` and `go.sum`
```

We don't use punctuation at the end of list items.

> 1. If the user has enabled the option `gomodTidy`, `gomodTidy1.17` or `gomodTidyE` in the [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions) array, then Renovate runs `go mod tidy`, `go mod tidy -compat=1.17` or `go mod tidy -e` respectively, which itself can update `go.mod` and `go.sum`

Reviewer: Do we _really_ want this big list of internal details? This one bullet point now has:

Three variants of the `go mod tidy` command:

- `go mod tidy`
- `go mod tidy -compat=1.17`
- `go mod tidy -compat=1.17`

Two file formats:

- `go.mod`
- `go.sum`

So we're showing five things in one bullet point. I find this hard to understand, and it's easy to lose track of the five things while reading.

Author: For me personally I am always a big fan of knowing which commands are executed and not having a "black box" from end-user perspective. But it's your project and your decision.  :)

Reviewer: Thank you for your thoughts on this! 😄

My point is: the current layout of the information is hard to read and understand. To summarize:

- If we decide this information is important for our users: create new issue to rewrite the section after this PR lands
- If this information is not key for our users: drop it

I'll let the maintainers decide what to do here. 😉

Reviewer 3: ```suggestion
1. If the user has any of the available `gomodTidy` options ( e.g. `gomodTidy1.17` ) in the [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions), then Renovate runs `go mod tidy` with the respective options ( multiple options are allowed ).
```
I think the mentioning of `go.mod` and `go.sum` is not necessary if someone wants to modify options they will know Go modules and if not they will `go mod tidy`. This will direct them to more sophisticated documentation regarding this than we can provide.

   1. `gomodTidy` is implicitly enabled for major updates if the user has enabled the option `gomodUpdateImportPaths` in the [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions) array. If go modules 1.17 compatibility is needed you need to explicitly set the option `gomodTidy1.17`.

>    1. If adding `gomodTidy1.17` and `gomodTidyE` at the same time, both options will be enabled and results in running `go mod tidy -compat=1.17 -e`.

Reviewer: ```suggestion
   1. If adding `gomodTidy1.17` and `gomodTidyE` at the same time, both options will be enabled and results in running `go mod tidy -compat=1.17 -e`
```

Same thing here.

>    1. `gomodTidy` is implicitly enabled for major updates if the user has enabled the option `gomodUpdateImportPaths` in the [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions) array. If go modules 1.17 compatibility is needed you need to explicitly set the option `gomodTidy1.17`
>    1. If adding `gomodTidy1.17` and `gomodTidyE` at the same time, both options will be enabled and results in running `go mod tidy -compat=1.17 -e`

Reviewer 3: ```suggestion
```
